packages/app/src/App.tsx

```diff
@@ -1,61 +1,50 @@
-import { css } from '@emotion/react';
-import { Box, CircularProgress } from '@mui/material';
-import { Suspense, useEffect, useRef, useState } from 'react';
+import { useEffect, useMemo } from 'react';
 
-import { Body } from './components/Body';
-import Header from './components/Header';
+import { UpsetProvenance, UpsetActions, getActions, initializeProvenanceTracking } from '@visdesignlab/upset2-react';
+import { useRecoilValue, useSetRecoilState } from 'recoil';
+import { dataSelector, encodedDataAtom } from './atoms/dataAtom';
+import { upsetConfigAtom } from './atoms/config/upsetConfigAtoms';
+import { Root } from './components/Root';
+import React from 'react';
 
 /** @jsxImportSource @emotion/react */
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
-const AppCss = css`
-  overflow: hidden;
-  height: 100vh;
-  display: grid;
-  grid-template-rows: min-content auto;
-`;
 
 function App() {
-  const ref = useRef<HTMLDivElement>(null);
-  const [headerHeight, setHeaderHeight] = useState(-1);
+  const multinetData = useRecoilValue(dataSelector);
+  const encodedData = useRecoilValue(encodedDataAtom);
+  const data = (encodedData === null) ? multinetData : encodedData
+  const config = useRecoilValue(upsetConfigAtom);
 
-  useEffect(() => {
-    const { current } = ref;
-    if (!current) return;
-
-    if (headerHeight > 0) return;
+  const setState = useSetRecoilState(upsetConfigAtom);
 
-    setHeaderHeight(current.clientHeight);
-  }, [headerHeight, ref]);
+  useEffect(() => {
+    setState(config);
+  }, [config, setState]);
+
+  const conf = useMemo(() => {
+    if (data !== null) {
+      const conf = {...config}
+      if (config.visibleSets.length === 0) {
+        const setList = Object.keys(data.sets);
+        conf.visibleSets = setList.slice(0, 6);
+      }
+
+      conf.visibleAttributes = data.attributeColumns.slice(0, 3);
```

Comment:
Does this overwrite the imported state?

Reply:
No. See this state imported with only one visible set:
![image](https://user-images.githubusercontent.com/35744963/231514943-52cf82b0-45fc-4604-a962-0605bbe498aa.png)

Reply:
Visible attributes is the `ReleaseDate, etc.` columns, right?

Reply:
This does not overwrite the attributes after importing.
I exported a state that had NO attributes visible.
Before importing the aforementioned state: 
![image](https://user-images.githubusercontent.com/35744963/231552956-cbd5a7a2-b426-4f9f-be65-3bb8b616db66.png)

After importing the state:
![image](https://user-images.githubusercontent.com/35744963/231553067-3d54b1bf-bdaf-4f1d-832e-2d1b32c1a270.png)